### PR TITLE
sum cool changes!!!

### PR DESCRIPTION
**Pull request by nexpid**
draft

### CODE REVIEW

File: src/lib/utils/showDecorationActionSheet.tsx [hunk 1]
```diff
@@ -28,6 +29,14 @@ export default (decoration: Decoration) =>
 			onClose: () => hideActionSheet()
 		},
 		options: [
+			{
+				icon: getAssetIDByName('ic_message_copy'),
+				label: 'Copy Decoration Hash',
+				onPress: () => {
+					clipboard.setString(decoration.hash);
+					showToast('Copied!', getAssetIDByName('toast_copy_message'));
```
Comment (nexpid) [inline src/lib/utils/showDecorationActionSheet.tsx]: what do i put here instead of the **Copied!** message

File: src/ui/components/Preset.tsx [hunk 2]
```diff
@@ -21,13 +22,40 @@ const styles = stylesheet.createThemedStyleSheet({
...
+					height: 16,
+					borderRadius: 8
+				}}
+			/>
+		);
+	}
+}
+
+// TODO do this the Fiery way (use like a store or something idk)
+const hasFetchedBefore = {};
```
Comment (nexpid) [inline src/ui/components/Preset.tsx]: is this okay

File: src/lib/utils/showDecorationActionSheet.tsx (hunk 1 above)
Comment (jackmthws) [reply src/lib/utils/showDecorationActionSheet.tsx]: "Copied Decoration Hash"

File: src/ui/components/Preset.tsx (hunk 2 above)
Comment (jackmthws) [reply src/ui/components/Preset.tsx]: this scares me

I would just not bother keeping track of if they've fetched before also why are you using fetchProfile? don't you need fetchUser

Comment (nexpid) [reply src/ui/components/Preset.tsx]: fetchProfile also fetches the user (fetchUser doesnt exist)

Comment (jackmthws) [reply src/ui/components/Preset.tsx]: https://github.com/Vendicated/Vencord/blob/a9e67e2955dfa07d734361c66ceabb0278ec5e34/src/webpack/common/utils.ts#L96
damn you discord. this module is probably findable on mobile with same props (this getUser actually gets it from the api not a store)

File: src/ui/components/Preset.tsx [hunk 3]
```diff
@@ -21,13 +22,36 @@ const styles = stylesheet.createThemedStyleSheet({
...
 // Make this return default avatars for null users
-function renderAvatar(user) {
-	if (user) return <Avatar user={user} size="size16" />
-	else return 
-};
+function renderAvatar({ user, id }) {
+	if (user) return <Avatar user={user} size="size16" />;
+	else {
+		const defaultAvatarIndex = Number((BigInt(id) >> 22n) % 6n);
+		console.log(id, defaultAvatarIndex);
```
Comment (jackmthws) [inline src/ui/components/Preset.tsx]: remove log

File: src/ui/components/Preset.tsx [hunk 4]
```diff
@@ -36,18 +60,30 @@ export default function Preset({ preset }: { preset: Preset }) {
...
+				<FormTitle
+					title={preset.name}
+					icon={
+						<SummarizedIconRow
+							iconWrapperStyle={styles.wrapper}
+							items={preset.authorIds.map((id) => {
+								const user = UserStore.getUser(id);
+
+								// Hopefully next time the user will be fetched
+								if (!user) UserUtils.getUser(id);
```
Comment (jackmthws) [inline src/ui/components/Preset.tsx:70]: you could make a react hook that takes a set of ids and rerenders after they're fetched

**Pull request merged**